django/utils/http.py

```diff
@@ -290,6 +290,8 @@ def is_safe_url(url, host=None):
         url = url.strip()
     if not url:
         return False
+    if six.PY2:
+        url = force_text(url, errors='replace')
```

Comment:
The control character issue is that `<a href="\x08//example.com">` will redirect to example.com but `is_safe_url()` would consider it safe before 011a54315e46acdf288003566b8570440f5ac985. 
In Firefox and Chrome, `<a href="\ufffd//example.com">` will be treated as path relative and redirect to something like `http://localhost:8000/page-where-link-appears/%EF%BF%BD//example.com`, so I don't see a problem.